corva/models/stream.py

```diff
@@ -62,14 +59,38 @@ def is_completed(self) -> bool:
         return False
 
 
-class StreamEvent(ListEvent[StreamEventData]):
+class StreamEvent(BaseEvent, StreamEventData):
     @staticmethod
-    def from_raw_event(event: str, **kwargs) -> StreamEvent:
-        app_key: str = kwargs['app_key']
+    def from_raw_event(event: str) -> List[StreamEvent]:
+        events = parse_raw_as(List[StreamEvent], event)
+
+        return events
+
+
+class StreamStateData(BaseData):
+    last_processed_timestamp: int = -1
+    last_processed_depth: float = -1
+
+
+class StreamContext(BaseContext[StreamEvent, StreamStateData]):
+    cache_data_cls: Type[StreamStateData] = StreamStateData
+    filter_by_timestamp: bool = False
+    filter_by_depth: bool = False
+
+    @cached_property
+    def event(self) -> StreamEvent:
+        from corva.utils import FilterStreamEvent
```

Comment:
let's move this import to the top

Reply:
deleted utils file, there were circular deps

Reply:
using local imports to resolve circular dependencies is a short-term fix. It will backfire later if we don't build the structure of the project right